compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -853,7 +853,11 @@ fn check_param_wf(tcx: TyCtxt<'_>, param: &hir::GenericParam<'_>) {
 
         // Const parameters are well formed if their type is structural match.
         hir::GenericParamKind::Const { ty: hir_ty, default: _ } => {
-            let ty = tcx.type_of(tcx.hir().local_def_id(param.hir_id));
+            let mut ty = tcx.type_of(tcx.hir().local_def_id(param.hir_id));
+            while let ty::Projection(ty::ProjectionTy { substs, item_def_id }) = ty.kind() {
```

Comment:
You should use a real normalization function. This doesn't work if the projection is nested inside an aggregate.

Reply:
Which one do you have in mind? The only one I know of is `bound_type_of`.

Reply:
that's not a normalization function XD

In wf-checking we use `enter_wf_checking_ctxt` to get a `wfcx` (though you'll need to pass in the actual body def id, don't use the one of the param). That wfcx should have a `normalize` method

Reply:
Oh. TIL :laughing:

Reply:
Done!